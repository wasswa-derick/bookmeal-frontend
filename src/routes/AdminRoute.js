import React from "react";
import { Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";

export const AdminRoute = ({
  isUserAuthenticated,
  isAdmin,
  component: Component,
  ...rest
}) => (
  <Route
    {...rest}
    render={props => {
      if (isUserAuthenticated) {
        return isAdmin ? <Component {...props} /> : <Redirect to="/" />;
      }
      return <Redirect to="/login" />;
    }}
  />
);

AdminRoute.propTypes = {
  isUserAuthenticated: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired,
  isAdmin: PropTypes.bool.isRequired
};

/**
 * @param {any} state
 * @returns {Object} props mapped from redux state
 */
function mapStateToProps(state) {
  return {
    isUserAuthenticated: !!state.authReducer.user.isLoggedIn,
    isAdmin: !!state.authReducer.user.isAdmin
  };
}

export default connect(mapStateToProps)(AdminRoute);
