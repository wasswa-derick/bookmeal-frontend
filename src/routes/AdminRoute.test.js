import React from "react";
import { shallow } from "enzyme";
import { AdminRoute } from "./AdminRoute";

describe("AdminRoute", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <AdminRoute
        isUserAuthenticated
        isAdmin={false}
        component={HTMLDivElement}
      />
    );
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
