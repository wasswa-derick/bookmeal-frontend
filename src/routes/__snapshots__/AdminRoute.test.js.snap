// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminRoute should render correctly 1`] = `
<Route
  render={[Function]}
/>
`;
